app/main/routes.py

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python3
+from app import mongo
+from flask import Blueprint, current_app, render_template
+
+main = Blueprint("main", __name__)
+
+# database collection variables
+desserts_collection = mongo.db.desserts
+recipes_collection = mongo.db.recipes
+
+#----- Global Helper -----#
+def get_total_recipes():
+        return int(recipes_collection.count())
+@main.context_processor
+def total_recipes():
+        return dict(total_recipes=get_total_recipes)
+
+
+#----- HOME -----#
+@main.route("/")
+def home():
+        carousel = recipes_collection.aggregate([{"$sample": {"size": 8}}])
+        
+        # get recipes for random recipe link
+        random_recipe = recipes_collection.aggregate([{"$sample": {"size": 1}}])

```

Comment:
DB calls are expensive, Instead of making another db call, pick one from carousel.

Reply:
Genius! One line of code now in Python, with minor tweaks in Jinja to get a |random item from the carousel instead! Thanks for that!